analyses/sv-analysis/02-shatterseek.R

```diff
@@ -0,0 +1,400 @@
+# choose independent specimens and remove all chrY
+# shatterseek can't recognize chrY, so remove them
+cnv_analysis <-  cnvconsensus[cnvconsensus$ID %in% bioid & cnvconsensus$chrom != "chrY",]
+# remove "chr", because shatterseek can't recognize it
+cnv_analysis$chrom <- gsub("chr","",cnv_analysis$chrom)
+
+
+## ===================== Run shatterseek in min.size = 6 or 3, combine and ouput the result =====================
+# ss6, ss3 and sv are used to merge data from different samples
+ss6 <- data.frame()
+ss3 <- data.frame()
+sv <- data.frame()
+
+# main process
+for (i in bioid) {
+  # read sv file one by one
+  sv_shatterseek <- read.table(file.path("scratch","sv-vcf",paste(i,"_withoutYandM.tsv",sep="")),sep="\t",header=TRUE)
+  
+  # sv_shatterseek_original is a file with chrY and ChrM, which will be used later
+  sv_shatterseek_original <- read.table(file.path("scratch","sv-vcf",paste(i,".tsv",sep="")),sep="\t",header=TRUE)
+  
+
+  
+  #  read cnv one by one
+  cnv_shatterseek <-  cnv_analysis[cnv_analysis$ID == i,]
+  
+  # For CI
+  # if we meet empty sv file, jump into next loop
+  nsv1 <- nrow(sv_shatterseek)
+  nsv2  <-  nrow(sv_shatterseek_original)
+  # if nsv1 or nsv2 is 0, jump into next loop
+  if (nsv1 == 0 | nsv2 == 0) {
+    print(paste0(i," has an empty sv file"))
+    next;
+  }
+  
+  
+  # add sample id
+  sv_shatterseek_original$sample <- i
+  # merge all sv_shatterseek_original, will be used later
+  sv <- rbind(sv,sv_shatterseek_original)
+  
+  
+  # build sv and cnv data frame
+  SV_data <-
+    SVs(
+      chrom1 = as.character(sv_shatterseek$chrom1),
+      pos1 = as.numeric(sv_shatterseek$pos1),
+      chrom2 = as.character(sv_shatterseek$chrom2),
+      pos2 = as.numeric(sv_shatterseek$pos2),
+      SVtype = as.character(sv_shatterseek$SVtype),
+      strand1 = as.character(sv_shatterseek$strand1),
+      strand2 = as.character(sv_shatterseek$strand2)
+    )
+  CN_data <-
+    CNVsegs(
+      chrom = as.character(cnv_shatterseek$chrom),
+      start = cnv_shatterseek$loc.start,
+      end = cnv_shatterseek$loc.end,
+      total_cn = cnv_shatterseek$copy.num
+    )
+  
+  # run shatterseek in min.size = 6
+  chromothripsis6 <- shatterseek(SV.sample=SV_data,seg.sample=CN_data,min.Size=6)
+  chrss6 <- chromothripsis6@chromSummary
+  chrss6$sample <- i
+  ss6 <- rbind(ss6,chrss6)
+  
+  # run shatterseek in min.size = 3
+  chromothripsis3 <- shatterseek(SV.sample=SV_data,seg.sample=CN_data,min.Size=3)
+  chrss3 <- chromothripsis3@chromSummary
+  chrss3$sample <- i
+  ss3 <- rbind(ss3,chrss3)
+}
+
+# save output
+write.csv(ss6,file.path(output_directory,"PBTA_shatterseek_call_minsize6.csv"),row.names = F)
+write.csv(ss3,file.path(output_directory,"PBTA_shatterseek_call_minsize3.csv"),row.names = F)
+write.csv(sv,file.path(output_directory,"PBTA_merge_sv.csv"),row.names=F)
+
+## ===================== Test and Filter The Combined Results, Then Ouput =====================
+# read files
+parkcall6 <- ss6
+parkcall3 <- ss3
```

Comment:
OK!

Reply:
> Because you're no longer reading this back in but are instead using the data frames that are already in the environment, could you alter the code above to assign the `shatterseek` output to `parkcall3` and `parkcall6` directly instead of replicating the object?

OK!